src/masternodes/mn_checks.cpp

```diff
@@ -3739,15 +3739,19 @@ bool IsDisabledTx(uint32_t height, CustomTxType type, const Consensus::Params& c
     // ICXCloseOffer       = '7',
 
     // Leaving close orders, as withdrawal of existing should be ok?
-    switch (type) {
-        case CustomTxType::ICXCreateOrder:
-        case CustomTxType::ICXMakeOffer:
-        case CustomTxType::ICXSubmitDFCHTLC:
-        case CustomTxType::ICXSubmitEXTHTLC:
-        case CustomTxType::ICXClaimDFCHTLC:
-            return true;
-        default:
-            return false;
+    // disable ICX orders for all networks other than testnet
+    if (Params().NetworkIDString() != CBaseChainParams::TESTNET)
+    {
+    	switch (type) {
+    	    case CustomTxType::ICXCreateOrder:
+    	    case CustomTxType::ICXMakeOffer:
+    	    case CustomTxType::ICXSubmitDFCHTLC:
+    	    case CustomTxType::ICXSubmitEXTHTLC:
+    	    case CustomTxType::ICXClaimDFCHTLC:
+    	        return true;
+    	    default:
+    	        return false;
+    	}
```

Comment:
```suggestion
    // disable ICX orders for all networks other than testnet
    if (Params().NetworkIDString() == CBaseChainParams::TESTNET) return false;
    
    switch (type) {
        case CustomTxType::ICXCreateOrder:
        case CustomTxType::ICXMakeOffer:
        case CustomTxType::ICXSubmitDFCHTLC:
        case CustomTxType::ICXSubmitEXTHTLC:
        case CustomTxType::ICXClaimDFCHTLC:
            return true;
        default:
            return false;
```

Fixes `warning: non-void function does not return a value in all control paths`